server/jetstream_api.go

```diff
@@ -1182,7 +1185,7 @@ func (s *Server) jsStreamSnapshotRequest(sub *subscription, c *client, subject,
 		return
 	}
 
-	sr, err := mset.Snapshot(0, !req.NoConsumers)
+	sr, err := mset.Snapshot(0, !req.NoConsumers, req.CheckMsgs)
```

Comment:
so this blocks and will eventually start yah? it could be a very long time right? Just to be sure I get the behaviour.  cli tool would just wait however long it takes?

other than that question and having to think that through LGTM

Reply:
Good point, maybe move this to the Go routine as well.

Reply:
Well on second thought I did it that way to make sure the caller got an errors from checking the stream.

Reply:
Yup we need that. I am just wondering how long it takes and what we show users then etc 

And how I/o intensive is wrt impact on other streams. We might not want more than 1 of these going on at a time perhaps?

Maybe better to make jsck a separate API even then we can easily do progress - send a msg seq number every second or something ?

Reply:
ok figured it out. Will fix.

Reply:
Let's get some mileage under our belt with it. Maybe separate API but can add later.